### ISSUE TITLE
Microsoft account logged out when unable to reach microsoft server

### ISSUE THREAD

**Issue by leElvyn** [modrinth/code#1082]
### Describe the bug

When launching theseus, if because of a connection issue or a proxy blocking xbox servers, you get logged out of your microsoft account on the next launch. Additionally, the launcher doesn't go in offline mode, and it is impossible to launch any instance. 

### Steps to reproduce

1. Be behind a proxy blocking the xbox / microsoft server 
2. Launch modrinth 
3. Try to launch an instance
4. Notice it is impossible to launch an instance
5. Restart Theseus
6. Notice the microsoft account has been removed

### Expected behavior

The launcher starts the instance in offline mode, and does not log out.

### System information

Tested on Fedora 39 and windows 11.

**Comment by ByteAfterlife** [comment]
Yep. That's my issue. I tested on Windows 10, 11, and Linux Mint 6.2.9

Modrinth version: 0.7.1

**Comment by Prospector** [comment]
This should no longer be an issue
